codalab_service.py

```diff
@@ -28,12 +31,15 @@ def print_header(description):
 
 def should_run_service(args, service):
     # `default` is generally used to bring up everything for local dev or quick testing.
-    # `default-no-worker` is generally used for real deployment since we don't
-    # want a worker running on the same machine.
-    return (
-        service in args.services
-        or (service != 'test' and 'default' in args.services)
-        or (service != 'test' and service != 'worker' and 'default-no-worker' in args.services)
+    # `default no-worker` is generally used for real deployment since we don't want a worker running on the same machine.
+    default_services = DEFAULT_SERVICES if 'default' in args.services else []
+    # remove "no-" prefix to get the list of services that will be excluded from execution
+    no_default_services = [svc.replace("no-", "") for svc in args.services]
```

Comment:
This doesn't seem right...because you only want to do it for services that contain `no-`.

I think a better way to do this would be:
- Let `services` be `args.services` with `default` expanded into `DEFAULT_SERVICES`.
- Return whether `service in services` and not `negate_service(service) in services`.